docs/backstage-guide-v1.md

```diff
@@ -0,0 +1,313 @@
+    "ghcr.io/devcontainers-contrib/features/typescript:2": {},
+    "ghcr.io/devcontainers/features/git:1": {}
+  },
+  "overrideFeatureInstallOrder": [
+    "ghcr.io/devcontainers/features/common-utils",
+    "ghcr.io/devcontainers/features/git",
+    "ghcr.io/devcontainers/features/node",
+    "ghcr.io/devcontainers-contrib/features/typescript"
+  ],
+  "runArgs": ["--network=host"],
+  "customizations": {
+    "vscode": {
+      "extensions": [
+        "esbenp.prettier-vscode",
+        "dbaeumer.vscode-eslint",
+        "streetsidesoftware.code-spell-checker",
+        "ms-vscode.vscode-typescript-next"
+      ]
+    }
+  },
+  "forwardPorts": [
+    3000,
+    7007
+  ],
+  "portsAttributes": {
+    "3000": {
+      "label": "Frontend",
+      "onAutoForward": "notify"
+    },
+    "7007": {
+      "label": "Backend",
+      "onAutoForward": "ignore"
+    }
+  },
+  "containerEnv": {
+    "HOST": "0.0.0.0"
+  },
+  "workspaceFolder": "/workspaces/starter-backstage",
+  "postCreateCommand": "yarn install",
+  "remoteUser": "daytona"
+}
+```
+This configuration includes:
+
+- **name**: Specifies the name of the development environment as "Backstage Dev Container."
+- **image**: Uses the `ubuntu:22.04` Docker image as the base for the development environment.
+- **features**: 
+  - **common-utils**: Adds common utilities with configurations for the user "daytona" (UID: 1000, GID: 1000) and sets Zsh as the default shell.
+  - **node**: Installs Node.js version 18 with dependencies for `node-gyp` and manages Node versions using NVM 0.39.5.
+  - **typescript**: Adds TypeScript support for the development environment.
+  - **git**: Installs Git to manage source code versioning.
+- **overrideFeatureInstallOrder**: Specifies the order of feature installation to ensure common utilities, Git, Node.js, and TypeScript are set up in the correct sequence.
+- **runArgs**: Uses the `--network=host` argument to ensure the container shares the host's network, simplifying network configurations for development.
+- **customizations**: Installs essential Visual Studio Code extensions, including Prettier, ESLint, a spell checker, and TypeScript Next.
+- **forwardPorts**: Sets up port forwarding for the frontend (3000) and backend (7007) servers.
+- **portsAttributes**: Labels the forwarded ports with "Frontend" for 3000 and "Backend" for 7007. The frontend port will notify on auto-forward, while the backend port will be ignored.
+- **containerEnv**: Sets the environment variable `HOST` to `0.0.0.0`, ensuring that the application is accessible on all network interfaces.
+- **workspaceFolder**: Sets the workspace folder to `/workspaces/starter-backstage` to match the local workspace folder name.
+- **postCreateCommand**: Runs `yarn install` to automatically install project dependencies when the container is created.
+- **remoteUser**: Sets "daytona" as the remote user for running commands within the container.
+
+## Building and Customizing Plugins in Backstage
```

Comment:
this should be mentioned in the intro, as all of the setup is leading to this

Reply:
sure, will ensure these changes are reflected in next commit.